### PR TITLE
feat: Gather clients expiry metrics for paths on github

### PR DESCRIPTION
**Pull request by kayano**
This changes the exporter to use only golang and provided rpc endpoints for getting ibc clients expiry metrics.

Instead relaying on rly binary and configured go relayer it goes now during startup to configured github repo, eg. https://github.com/archway-network/networks/tree/main/_IBC and fetches ibc paths configuration from there.

Using configured rpc endpoints for fetched paths it gets `cosmos_ibc_client_expiry` metric for each ibc client in those paths.

### CODE REVIEW

File: pkg/collector/collector.go [hunk 1]
```diff
@@ -1,88 +1,54 @@
...
 }
 
-func (rc RelayerCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- up
-}
+func (cc IBCClientsCollector) Describe(_ chan<- *prometheus.Desc) {}
 
-func (rc RelayerCollector) Collect(ch chan<- prometheus.Metric) {
-	log.Debug("Start collecting", zap.String("metric", configuredChainMetricName))
+func (cc IBCClientsCollector) Collect(ch chan<- prometheus.Metric) {
```
Comment (shahbazn) [inline pkg/collector/collector.go:30]: do we need to explicityly initilize this?

File: pkg/ibc/ibc.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+	clientsInfo.ChainBClientExpiration, clientsInfo.ChainBClientInfo, err = relayer.QueryClientExpiration(ctx, chainB, chainA)
+	if err != nil {
+		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdB, cdA)
+	}
+
+	return clientsInfo, nil
+}
+
+func (c *Config) IBCPaths() ([]*relayer.IBCdata, error) {
```
Comment (shahbazn) [inline pkg/ibc/ibc.go:156]: Lets separate out config management into a separat package to keep ibc package clean

File: cmd/relayer_exporter/relayer_exporter.go [hunk 3]
```diff
@@ -36,8 +37,24 @@ func main() {
...
-	rc := collector.RelayerCollector{Rly: *rly}
-	prometheus.MustRegister(rc)
+	cfg, err := ibc.NewConfig(*configPath)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	log.Info(fmt.Sprintf("Getting IBC paths from %s/%s/%s on GitHub", cfg.GitHub.Org, cfg.GitHub.Repo, cfg.GitHub.IBCDir))
+
+	paths, err := cfg.IBCPaths()
```
Comment (shahbazn) [inline cmd/relayer_exporter/relayer_exporter.go:47]: TODO to be added hear to poll for new paths at a configurable frequency

File: pkg/ibc/ibc.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+	return fmt.Sprintf("%s<->%s", chainAID, chainBID)
+}
+
+type RPC struct {
+	ChainID string `yaml:"chainId"`
+	URL     string `yaml:"url"`
+}
+
+type Config struct {
```
Comment (shahbazn) [inline pkg/ibc/ibc.go:59]: Perhaps a separate package for config?

File: pkg/collector/collector.go [hunk 5]
```diff
@@ -1,88 +1,54 @@
...
-	configuredChain = prometheus.NewDesc(
-		configuredChainMetricName,
-		"Returns configured chain.",
-		[]string{"chain_id"}, nil,
-	)
+var clientExpiry = prometheus.NewDesc(
+	clientExpiryMetricName,
+	"Returns light client expiry in unixtime.",
+	[]string{"chain_id", "client_id", "path"}, nil,
 )
```
Comment (shahbazn) [inline pkg/collector/collector.go:21]: perhaps using 'host_chain_id instead' of 'chain_id' and 'target_chain_id instead' of 'path' would be more explicit and descriptive here.

**Review by shahbazn — COMMENTED** [review]
Tested with default configs works well. Added a few comments regarding code structure and acknowleging future work.

File: pkg/collector/collector.go (hunk 5 above)
Comment (kayano) [reply pkg/collector/collector.go:21]: What about labels like `chain_id`, `client_id` and `target_chain_id`?  With `host_chain_id` I'm not sure it is obvious that `client_id` is for client on `host_chain_id`

**Review by shahbazn — APPROVED** [review]
lgtm

**Pull request merged**